src/AiClient.php

```diff
@@ -114,14 +115,44 @@ public static function defaultRegistry(): ProviderRegistry
     /**
      * Checks if a provider is configured and available for use.
      *
+     * Supports multiple input formats for developer convenience:
+     * - ProviderAvailabilityInterface: Direct availability check (backward compatible)
+     * - string (provider ID): e.g., AiClient::isConfigured('openai')
+     * - string (class name): e.g., AiClient::isConfigured(OpenAiProvider::class)
+     *
+     * When using string input, this method leverages the ProviderRegistry's centralized
+     * dependency management, ensuring HttpTransporter and authentication are properly
+     * injected into availability instances.
+     *
      * @since 0.1.0
      *
-     * @param ProviderAvailabilityInterface $availability The provider availability instance to check.
+     * @param ProviderAvailabilityInterface|string|class-string<ProviderInterface> $availabilityOrIdOrClassName
+     *        The provider availability instance, provider ID, or provider class name.
+     * @param ProviderRegistry|null $registry Optional custom registry. If null, uses default.
      * @return bool True if the provider is configured and available, false otherwise.
      */
-    public static function isConfigured(ProviderAvailabilityInterface $availability): bool
+    public static function isConfigured($availabilityOrIdOrClassName, ?ProviderRegistry $registry = null): bool
```

Comment:
This shouldn't allow passing a custom registry, that's unrelated. The `AiClient` for now should always operate with the `defaultRegistry()`.